compiler errors, invalid conversion, no member named channels

### Steps to reproduce

Try to build `telegram-desktop-git` from AUR on Arch Linux in clean chroot.

### Expected behaviour

Should build without error.  5.6.3 builds successfully.

### Actual behaviour

Build stops at 720/1513.

```
/home/builder/build/src/telegram-desktop/Telegram/SourceFiles/ui/controls/round_video_recorder.cpp: In member function ‘void Ui::RoundVideoRecorder::Private::initEncoding()’:
/home/builder/build/src/telegram-desktop/Telegram/SourceFiles/ui/controls/round_video_recorder.cpp:273:17: error: invalid conversion from ‘int (*)(void*, uint8_t*, int)’ {aka ‘int (*)(void*, unsigned char*, int)’} to ‘int (*)(void*, const uint8_t*, int)’ {aka ‘int (*)(void*, const unsigned char*, int)’} [-fpermissive]
  273 |                 &Private::Write,
      |                 ^~~~~~~~~~~~~~~
      |                 |
      |                 int (*)(void*, uint8_t*, int) {aka int (*)(void*, unsigned char*, int)}
In file included from /home/builder/build/src/telegram-desktop/Telegram/SourceFiles/ui/controls/round_video_recorder.cpp:12:
/home/builder/build/src/telegram-desktop/Telegram/SourceFiles/ffmpeg/ffmpeg_utility.h:147:14: note:   initializing argument 3 of ‘FFmpeg::FormatPointer FFmpeg::MakeWriteFormatPointer(void*, int (*)(void*, uint8_t*, int), int (*)(void*, const uint8_t*, int), int64_t (*)(void*, int64_t, int), const QByteArray&)’
  147 |         int(*write)(void *opaque, const uint8_t *buffer, int bufferSize),
      |         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/builder/build/src/telegram-desktop/Telegram/SourceFiles/ui/controls/round_video_recorder.cpp: In member function ‘bool Ui::RoundVideoRecorder::Private::initAudio()’:
/home/builder/build/src/telegram-desktop/Telegram/SourceFiles/ui/controls/round_video_recorder.cpp:390:22: error: ‘struct AVCodecContext’ has no member named ‘channels’
  390 |         _audioCodec->channels = _audioCodec->ch_layout.nb_channels;
      |                      ^~~~~~~~
/home/builder/build/src/telegram-desktop/Telegram/SourceFiles/ui/controls/round_video_recorder.cpp: In member function ‘Ui::RoundVideoResult Ui::RoundVideoRecorder::Private::appendToPrevious(Ui::RoundVideoResult)’:
/home/builder/build/src/telegram-desktop/Telegram/SourceFiles/ui/controls/round_video_recorder.cpp:517:17: error: invalid conversion from ‘int (*)(void*, uint8_t*, int)’ {aka ‘int (*)(void*, unsigned char*, int)’} to ‘int (*)(void*, const uint8_t*, int)’ {aka ‘int (*)(void*, const unsigned char*, int)’} [-fpermissive]
  517 |                 &Private::Write,
      |                 ^~~~~~~~~~~~~~~
      |                 |
      |                 int (*)(void*, uint8_t*, int) {aka int (*)(void*, unsigned char*, int)}
/home/builder/build/src/telegram-desktop/Telegram/SourceFiles/ffmpeg/ffmpeg_utility.h:147:14: note:   initializing argument 3 of ‘FFmpeg::FormatPointer FFmpeg::MakeWriteFormatPointer(void*, int (*)(void*, uint8_t*, int), int (*)(void*, const uint8_t*, int), int64_t (*)(void*, int64_t, int), const QByteArray&)’
  147 |         int(*write)(void *opaque, const uint8_t *buffer, int bufferSize),
      |         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/builder/build/src/telegram-desktop/Telegram/SourceFiles/ui/controls/round_video_recorder.cpp: In member function ‘bool Ui::RoundVideoRecorder::Private::copyPackets(gsl::not_null<AVFormatContext*>, gsl::not_null<AVFormatContext*>, CopyContext&, crl::time)’:
/home/builder/build/src/telegram-desktop/Telegram/SourceFiles/ui/controls/round_video_recorder.cpp:584:23: warning: ‘void av_init_packet(AVPacket*)’ is deprecated [-Wdeprecated-declarations]
  584 |         av_init_packet(&packet);
      |         ~~~~~~~~~~~~~~^~~~~~~~~
In file included from /usr/include/libavcodec/avcodec.h:44,
                 from /home/builder/build/src/telegram-desktop/Telegram/SourceFiles/ffmpeg/ffmpeg_utility.h:19:
/usr/include/libavcodec/packet.h:653:6: note: declared here
  653 | void av_init_packet(AVPacket *pkt);
      |      ^~~~~~~~~~~~~~
```
</details>


### Operating system

Arch Linux

### Version of Telegram Desktop

5.6.4: 78937d716fe6e8fa3b470d378d2fcb54a643f06c

### Installation source

Other (unofficial) source

### Crash ID

_No response_

### Logs

[`telegram-desktop-git.log`](https://github.com/user-attachments/files/17534892/telegram-desktop-git.log)

--- a/Telegram/SourceFiles/ui/controls/round_video_recorder.cpp
+++ b/Telegram/SourceFiles/ui/controls/round_video_recorder.cpp
@@ -107,9 +107,15 @@ private:
 		std::array<int64, kMaxStreams> lastDts = { 0 };
 	};
 
+#if DA_FFMPEG_CONST_WRITE_CALLBACK
+	static int Write(void *opaque, const uint8_t *_buf, int buf_size) {
+		uint8_t *buf = const_cast<uint8_t *>(_buf);
+#else
 	static int Write(void *opaque, uint8_t *buf, int buf_size) {
+#endif
 		return static_cast<Private*>(opaque)->write(buf, buf_size);
 	}
+
 	static int64_t Seek(void *opaque, int64_t offset, int whence) {
 		return static_cast<Private*>(opaque)->seek(offset, whence);
 	}
@@ -388,7 +394,6 @@ bool RoundVideoRecorder::Private::initAudio() {
 	_audioCodec->sample_rate = kAudioFrequency;
 #if DA_FFMPEG_NEW_CHANNEL_LAYOUT
 	_audioCodec->ch_layout = AV_CHANNEL_LAYOUT_MONO;
-	_audioCodec->channels = _audioCodec->ch_layout.nb_channels;
 #else
 	_audioCodec->channel_layout = AV_CH_LAYOUT_MONO;
 	_audioCodec->channels = _audioChannels;

